pkg/cmd/gist/list/list_test.go

```diff
@@ -0,0 +1,217 @@
+			cli:  "--secret",
+			wants: ListOptions{
+				Limit:      10,
+				Visibility: "secret",
+			},
+		},
+		{
+			name: "public and secret",
+			cli:  "--secret --public",
+			wants: ListOptions{
+				Limit:      10,
+				Visibility: "all",
+			},
+		},
+		{
+			name: "limit",
+			cli:  "--limit 30",
+			wants: ListOptions{
+				Limit:      30,
+				Visibility: "all",
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			f := &cmdutil.Factory{}
+
+			argv, err := shlex.Split(tt.cli)
+			assert.NoError(t, err)
+
+			var gotOpts *ListOptions
+			cmd := NewCmdList(f, func(opts *ListOptions) error {
+				gotOpts = opts
+				return nil
+			})
+			cmd.SetArgs(argv)
+			cmd.SetIn(&bytes.Buffer{})
+			cmd.SetOut(&bytes.Buffer{})
+			cmd.SetErr(&bytes.Buffer{})
+
+			_, err = cmd.ExecuteC()
+			assert.NoError(t, err)
+
+			assert.Equal(t, tt.wants.Visibility, gotOpts.Visibility)
+			assert.Equal(t, tt.wants.Limit, gotOpts.Limit)
+		})
+	}
+}
+
+func Test_listRun(t *testing.T) {
+	tests := []struct {
+		name    string
+		opts    *ListOptions
+		wantOut string
+		stubs   func(*httpmock.Registry)
+		nontty  bool
+	}{
+		{
+			name: "no gists",
+			opts: &ListOptions{},
+			stubs: func(reg *httpmock.Registry) {
+				reg.Register(httpmock.REST("GET", "gists"),
+					httpmock.JSONResponse([]shared.Gist{}))
+
+			},
+			wantOut: "",
+		},
+		{
+			name:    "default behavior",
+			opts:    &ListOptions{},
+			wantOut: "1234567890  cool.txt              1 file    public  about 6 hours ago\n4567890123                        1 file    public  about 6 hours ago\n2345678901  tea leaves thwart...  2 files   secret  about 6 hours ago\n3456789012  short desc            11 files  secret  about 6 hours ago\n",
+		},
+		{
+			name:    "with public filter",
+			opts:    &ListOptions{Visibility: "public"},
+			wantOut: "1234567890  cool.txt  1 file  public  about 6 hours ago\n4567890123            1 file  public  about 6 hours ago\n",
+		},
+		{
+			name:    "with secret filter",
+			opts:    &ListOptions{Visibility: "secret"},
+			wantOut: "2345678901  tea leaves thwart...  2 files   secret  about 6 hours ago\n3456789012  short desc            11 files  secret  about 6 hours ago\n",
+		},
+		{
+			name:    "with limit",
+			opts:    &ListOptions{Limit: 1},
+			wantOut: "1234567890  cool.txt  1 file  public  about 6 hours ago\n",
+		},
+		{
+			name:    "nontty output",
+			opts:    &ListOptions{},
+			wantOut: "1234567890\tcool.txt\t1 file\tpublic\t0001-01-01 00:00:00 +0000 UTC\n4567890123\t\t1 file\tpublic\t0001-01-01 00:00:00 +0000 UTC\n2345678901\ttea leaves thwart those who court catastrophe\t2 files\tsecret\t0001-01-01 00:00:00 +0000 UTC\n3456789012\tshort desc\t11 files\tsecret\t0001-01-01 00:00:00 +0000 UTC\n",
+			nontty:  true,
+		},
+	}
+
+	for _, tt := range tests {
+		reg := &httpmock.Registry{}
+		if tt.stubs == nil {
+			reg.Register(httpmock.REST("GET", "gists"),
+				httpmock.JSONResponse([]shared.Gist{
```

Comment:
Would it be easier to add the `UpdatedAt` field here to each of the gists so that we can remove the `ListOptions.Since` function? It seems like that was added for testing purposes mostly.

Reply:
good idea, updated